src/porepy/numerics/ad/operators.py

```diff
@@ -190,72 +287,25 @@ def set_name(self, name: str) -> None:
         """
         self._name = name
 
-    def previous_timestep(self) -> pp.ad.Operator:
-        """Return an operator that represents the value of this operator at the previous
-        timestep.
-
-        The operator tree at the previous time step is created as a shallow copy, and will
-        thus be identical to the original operator, except that all time dependent operators
-        are evaluated at the previous time step.
+    def previous_timestep(self, steps: int = 1) -> pp.ad.Operator:
+        """Base method to trigger a recursion over the operator tree and create a
```

Comment:
Because Operator classes inheriting from `TimeDependentOperator` or `IterativeOperator` are overriding this method implementing the shift.
They are leaves in these recursions.

Reply:
I see. As a follow-up question: Say I have
```
a: TimeDependentOperator = ...
b: TimeDependentOperator = ...
c: Operator = ... # (not TimeDependentOperator)
d = a * b
e = a + c
```
Are any of d and e `TimeDependentOperator`?

Reply:
d and e are created in the arithmetic overloads of `Operator`, therefore they are basic operators with a defined `Operation` and have no time or iterate index per se. One of their children has them though, and this has an effect when the operator is parsed.
d and e do not represent leaves in the operator tree, they are operations in the sense of the word operation.
Where as `a` and `b` are leaves and do not represent an operation, but an term.

Reply:
Thanks, that's what I thought. I just wanted to be sure I hadn't missed something.